runtime/citrus-xml/src/test/java/org/citrusframework/integration/ShardedXmlTestLoaderIT.java

```diff
@@ -0,0 +1,41 @@
+package org.citrusframework.integration;
+
+import org.citrusframework.annotations.CitrusTestSource;
+import org.citrusframework.testng.TestNGCitrusSupport;
+import org.testng.annotations.Test;
+
+import static org.citrusframework.common.ShardedTestLoaderAndExecutor.XML_SHARDED;
+
+@Test
+public class ShardedXmlTestLoaderIT extends TestNGCitrusSupport {
+
+    @CitrusTestSource(type = XML_SHARDED, sources = {"classpath:org/citrusframework/integration"})
+    public void loadFromSources() {
+        // TODO: Check if test has been executed
```

Comment:
question @christophd, how would you do this? how can I check how many tests have actually been executed? e.g. here it would be one, in the recursive tests it would be three?

Reply:
maybe add a custom test reporter (or test listener) that counts the number of executed tests. and after the test run verify the tests that reported success/failure in the custom reporter.